algorithms_task_8/student/src/impl.cpp

```diff
@@ -1,10 +1,25 @@
 #include "impl.h"
+using namespace std;
+using row = std::pair<CVGroup, keywords>;
 
 skillMatrix buildSkillsMatrix(const keywords &cvWords, const skillMatrix &generalMatrix)
 {
     skillMatrix cvMatrix;
 
     // convert from general matrix to cv matrix by cvWords
+    transform(generalMatrix.begin(), generalMatrix.end(),
+              std::inserter(cvMatrix, cvMatrix.begin()) ,
+              [&cvWords](row const & skillRow) {
+                  keywords skillKeywords;
+                  //still don't understand about set algorithms usability in this case
```

Comment:
So, you are doing next steps:
1. You check if element from first range exists in second range
2. If it exists, you add it to 3rd range

Isn't it a intersection of two ranges?